src/main/kotlin/org/rust/lang/core/psi/ext/RsPathReferenceElement.kt

```diff
@@ -0,0 +1,12 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.psi.ext
+
+import org.rust.lang.core.resolve.ref.RsPathReference
+
+interface RsPathReferenceElement : RsReferenceElement{
```

Comment:
And do we really need this class?

Reply:
I think we need it at the moment, though I still think that perhaps we should dramatically reduce the number of reference classes and separate "references for integration with the platform"  from "Rust-specific name resolution".